### PR TITLE
Refactor explainer

### PR DESCRIPTION
**Pull request by klausw**
Simplify introduction, remove redundant parts and/or move details
to an "implementation details" section, add a clearer overview,
and update text and examples to be easier to follow.

### CODE REVIEW

File: explainer.md [hunk 1]
```diff
@@ -19,51 +17,143 @@ A smartphone AR game could show onscreen buttons for game controls that are not
...
+For headset-based AR where the DOM overlay is displayed as a floating rectangle along with a tracked motion controller. DOM UI interactions would typically be based on a ray emanating from the controller. If the same controller and primary trigger are also used for XR world interactions, the application needs to be able to disambiguate the input. For example, if the application uses the primary trigger to activate a sculpting tool, this should be temporarily suppressed if the user is trying to click on a button shown in the DOM overlay.
```
Comment (bialpio) [inline explainer.md:154]:
```suggestion
For headset-based AR where the DOM overlay is displayed as a floating rectangle along with a tracked motion controller, DOM UI interactions would typically be based on a ray emanating from the controller. If the same controller and primary trigger are also used for XR world interactions, the application needs to be able to disambiguate the input. For example, if the application uses the primary trigger to activate a sculpting tool, this should be temporarily suppressed if the user is trying to click on a button shown in the DOM overlay.
```

File: explainer.md [hunk 2]
```diff
@@ -73,58 +163,48 @@ document.getElementById('ui-container').addEventListener('beforexrselect', (ev)
...
+WebXR also supports non-event-based input. This includes controller poses, button/axis states, and transient XR input sources such as [Screen-based input](https://github.com/immersive-web/webxr/blob/master/input-explainer.md#screen) that creates 3D targeting rays based on 2D screen touch points. These inputs are not affected by DOM overlays and continue to be processed as usual. Applications are responsible for deduplicating these non-event inputs if they overlap with DOM events, though it's recommended to avoid UI designs that depend on this since there are many ambiguous corner cases, for example pointer movement that starts on a DOM element and ends outside it.
```
Comment (bialpio) [inline explainer.md:166]: The change in this paragraph makes the last sentence a bit long.

File: explainer.md (hunk 1 above)
Comment (klausw) [reply explainer.md:154]: Done.

File: explainer.md (hunk 2 above)
Comment (klausw) [reply explainer.md:166]: Reverting this change.

**Pull request merged**